src/test_fixture/logging.rs

```diff
@@ -1,16 +1,29 @@
 use metrics_tracing_context::MetricsLayer;
-use std::sync::Once;
+use std::{str::FromStr, sync::Once};
 use tracing::Level;
-use tracing_subscriber::{fmt, layer::SubscriberExt, util::SubscriberInitExt, EnvFilter};
+use tracing_subscriber::{
+    filter::Directive, fmt, layer::SubscriberExt, util::SubscriberInitExt, EnvFilter,
+};
 
+/// Set up logging for IPA
+///
+/// ## Panics
+/// Does not, but compiler cannot be convinced otherwise
 pub fn setup() {
     static INIT: Once = Once::new();
 
     INIT.call_once(|| {
+        let default_directive = if let Some(crate_name) = option_env!("CARGO_CRATE_NAME") {
+            // only print IPA crate logging by default
+            Directive::from_str(&format!("{crate_name}=INFO")).unwrap()
```

Comment:
https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/targets/struct.Targets.html#method.with_target seems like a better way to get a directive without having to build and parse a string.

Reply:
Unfortunately `with_target` statically configures the logger and does not allow `RUST_LOG` to override it. [`with_default` ](https://docs.rs/tracing-subscriber/latest/tracing_subscriber/filter/targets/struct.Targets.html#method.with_default) looks almost what we would want but it only takes the level argument which would again print Shuttle output

Reply:
Oh, I see what is going on now.  I am referring to a `Filter`, which is not a `Directive`.  A `Directive` is not a real filter, it's an intermediate format.  I don't like how that crate works now.  I shouldn't have looked at the code.

Reply:
yea, I wasn't too concerned about it because this crate provides the right abstraction for layers and filters, so we can implement our own layer that does the filtering the way we like